variables.tf

```diff
@@ -1,36 +1,109 @@
-
-#
-# Developer tips:
-#   - Below are some common module input variables
-#   - They should be updated for input variables applicable to the module being added
-#   - Use variable validation when possible
-#
-
-variable "name" {
-  type        = string
-  description = "A descriptive name used to identify the resource instance."
+########################################################################
+# Activity Tracker Event Routing
+#########################################################################
+
+# COS Targets
+variable "cos_targets" {
+  type = list(object({
+    endpoint                          = string
+    bucket_name                       = string
+    instance_id                       = string
+    api_key                           = optional(string)
+    service_to_service_enabled        = optional(bool, true)
+    target_region                     = optional(string)
+    target_name                       = string
+    skip_atracker_cos_iam_auth_policy = optional(bool, false)
+  }))
+  default     = []
+  description = "List of cos target to be created"
+  sensitive   = true
+
+  validation {
+    condition     = alltrue([for cos_target in var.cos_targets : (cos_target.service_to_service_enabled == true && cos_target.api_key == null) || (cos_target.service_to_service_enabled == false && cos_target.api_key != null)])
+    error_message = "If 'service_to_service_enabled' is true, 'api_key' value should not be passed. If you wish to use 'api_key', set 'service_to_service_enabled' to false."
+  }
 }
 
-variable "plan" {
-  type        = string
-  description = "The name of the plan type supported by service."
-  default     = "standard"
+# Event Streams Targets
+variable "eventstreams_targets" {
+  type = list(object({
+    instance_id                      = string
+    brokers                          = list(string)
+    topic                            = string
+    api_key                          = optional(string)
+    service_to_service_enabled       = optional(bool, true)
+    target_region                    = optional(string)
+    target_name                      = string
+    skip_atracker_es_iam_auth_policy = optional(bool, false)
+  }))
+  default     = []
+  description = "List of event streams target to be created"
+  sensitive   = true
+
   validation {
-    condition     = contains(["standard", "cos-one-rate-plan"], var.plan)
-    error_message = "The specified pricing plan is not available. The following plans are supported: 'standard', 'cos-one-rate-plan'"
+    condition     = alltrue([for es_target in var.eventstreams_targets : (es_target.service_to_service_enabled == true && es_target.api_key == null) || (es_target.service_to_service_enabled == false && es_target.api_key != null)])
+    error_message = "If 'service_to_service_enabled' is true, 'api_key' value should not be passed. If you wish to use 'api_key', set 'service_to_service_enabled' to false."
```

Comment:
Its the same message as above. I am not sure if we need to tell user if this error message is coming for ES target or COS. Recheck and change if required.